### PR TITLE
Add more tests, various fixes, separate out market maker stuff into standalone repo

### PR DESCRIPTION
**Pull request by InfiniteStyles**
I think that when the payout and outcomeToken amount is too small, i.e: (1/1000) payout, and 1 collateral token, it divides it into a number that's less than 1, and our system breaks. 

uint payout = otAmount.mul(payoutNumerator).div(payoutDenominator[outcomeTokenSetId]);

(1outcomeCount*1payout)/1000denominator = .0010

### CODE REVIEW

**Comment by cag** [comment]
Not a bug, just a consequence of rounding. Most discrepencies will be on the wei level, so not a big deal.

**Comment by cag** [comment]
Reopened because PR is wider in scope than message indicated.

**Comment by InfiniteStyles** [comment]
Updated this and checked the gas stats for both versions. redeemPayout gas usage has improved. Should be good now.

File: contracts/Events/EventManager.sol [hunk 1]
```diff
@@ -47,8 +47,6 @@ contract EventManager is OracleConsumer {
                 payout := calldataload(add(0x64, mul(0x20, i)))
             }
             payoutDenominator[outcomeTokenSetId] = payoutDenominator[outcomeTokenSetId].add(payout);
-
-            require(payoutForOutcomeToken[outcomeTokens[outcomeTokenSetId][i]] == 0, "payout already set");
```
Comment (cag) [inline contracts/Events/EventManager.sol]: Why remove this?

File: contracts/Events/EventManager.sol [hunk 2]
```diff
@@ -87,15 +86,15 @@ contract EventManager is OracleConsumer {
...
             if(otAmount > 0) {
                 ot.burnFrom(msg.sender, otAmount);
-                if(payoutNumerator > 0) {
-                    uint payout = otAmount.mul(payoutNumerator)
-                        .div(payoutDenominator[outcomeTokenSetId]);
-                    require(collateralToken.transfer(msg.sender, payout),
-                        "could not transfer payout to message sender");
+                if (payoutNumerator > 0) {
+                    uint payout = otAmount.mul(payoutNumerator).div(payoutDenominator[outcomeTokenSetId]);
                     totalPayout += payout;
```
Comment (cag) [inline contracts/Events/EventManager.sol]: Use the SafeMath version here maybe?

```solidity
totalPayout = totalPayout.add(payout);
```

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@gnosis.pm/gnosis-core-contracts",
-  "version": "1.0.0",
+  "name": "@gnosis.pm/pm-contracts",
```
Comment (cag) [inline package.json]: Why is the coloring so wonky???

File: contracts/MarketMakers/LMSRMarketMaker.sol [hunk 4]
```diff
@@ -1,132 +0,0 @@
-pragma solidity ^0.4.24;
```
Comment (cag) [inline contracts/MarketMakers/LMSRMarketMaker.sol]: Should these be removed yet?

**Pull request merged**